src/shared/jwt.js

```diff
@@ -0,0 +1,90 @@
+const jwt = require('jsonwebtoken');
+const config = require('../config');
+
+/**
+ * Validates user api JWT.
+ *
+ * If token is invalid (cannot be decoded): return status 500 and error
+ * If token is not in header: return status 401
+ */
+let validateUserToken = (req, res, next) => {
+    let token = req.headers['ha-api-token'];
+
+    if (token) {
+        jwt.verify(token, config.userPublicKey, config.verifyOptions, (err) => {
+            if (err)
+                res.status(500).send(err);
+
+            next();
+        });
+    }
+
+    return res.status(401).send();
+};
+
+/**
+ * Validates admin api JWT.
+ *
+ * If token is invalid (cannot be decoded): return status 500 and error
+ * If token is not in header: return status 401
+ */
+let validateAdminToken = (req, res, next) => {
+    let token = req.headers['ha-admin-token'];
+
+    if (token) {
+        jwt.verify(token, config.adminPublicKey, config.verifyOptions, (err, decoded) => {
+            if (err)
+                res.status(500).send(err);
+
+            next();
+        });
+    }
+
+    return res.status(401).send();
+};
+
+/**
+ * Validates user API request by decoding JWT and comparing decoded UUID against query param UUID.
+ * If token is invalid (cannot be decoded): return status 500 and error
+ *
+ * If token is invalid (cannot be decoded): return status 500 and error
+ * If token is not in header: return status 401
+ */
+let validateUserRequest = (req, res, next) => {
```

Comment:
Should this be `validateAdminRequest`, since where the token is stored in the header is different (`ha-admin-token` instead of `ha-api-token`)? Also, I know comparing `decoded.uid === req.query.uid` is important for some routes—which ones are you planning on using this function for?

Reply:
Actually, this should use `ha-api-token`! 

This is intended to validate user requests that target user data - e.g. if a user makes a request to update user information, this can determine that the `uid` associated with the table they're targeting is the same one encoded in their `ha-api-token`.

Reply:
Gotcha—sounds good! 👍

Reply:
Hold up! Then the below token should be `ha-api-token`!